kinto_fxa/scripts/process_account_events.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import uuid
+
+import boto3
+from ec2_metadata import ec2_metadata
+from kinto.core.utils import hmac_digest
+import transaction as current_transaction
+
+logger = logging.getLogger(__name__)
+
+
+def get_default_bucket_id(config, uid):
+    secret = config['registry'].settings['userid_hmac_secret']
+    digest = hmac_digest(secret, uid)
+    return str(uuid.UUID(digest[:32]))
+
+
+def process_account_events(config, queue_name, aws_region=None,
+                           queue_wait_time=20):
+    """Process account events from an SQS queue.
+
+    This function polls the specified SQS queue for account-related events,
+    processing each as it is found.  It polls indefinitely and does not return;
+    to interrupt execution you'll need to e.g. SIGINT the process.
+    """
+    logger.info("Processing account events from %s", queue_name)
+    try:
+        # Connect to the SQS queue.
+        # If no region is given, infer it from the instance metadata.
+        if aws_region is None:
+            logger.debug("Finding default region from instance metadata")
+            aws_region = ec2_metadata.region
+
+        logger.debug("Connecting to queue %r in %r", queue_name, aws_region)
+        sqs = boto3.resource('sqs', region_name=aws_region)
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+
+        # Poll for messages indefinitely.
+        # Use a wacky looping construct that can be mocked in tests.
+        for x in itertools.count():
+            msgs = queue.receive_messages(WaitTimeSeconds=queue_wait_time)
+            for msg in msgs:
+                process_account_event(config, msg.body)
+                # This intentionally deletes the event even if it was some
+                # unrecognized type.  No point leaving a backlog.
+                msg.delete()
+
+    except Exception:
+        logger.exception("Error while processing account events")
+        raise
+
+
+def process_account_event(config, body):
+    """Parse and process a single account event."""
+    registry = config['registry']
+    storage = registry.storage
+    permission = registry.permission
+
+    # Try very hard not to error out if there's junk in the queue.
+    try:
+        # Messages are a string of JSON, which, when parsed, has a
+        # Message field, which is a string of JSON that actually
+        # contains what we want.
+        event = json.loads(body)
+        event = json.loads(event['Message'])
+        event_type = event["event"]
+        uid = event["uid"]
+    except (ValueError, KeyError) as e:
+        logger.exception("Invalid account message: %r", e)
+    else:
+        if event_type == "delete":
+            # Delete everything from storage and permissions for
+            # this user.
+            logger.info("Processing account delete for %r", uid)
+            # FIXME: actually compute prefix correctly using
+            # config instead of just hardcoding fxa
+            uid = 'fxa:{}'.format(uid)
```

Comment:
I decided this wasn't important enough to implement right now because of the complexity of the INI file format for authentication policies. I could implement this now, later, or not at all, according to your preference.

Reply:
So just double-checking, both Notes and kinto-we use the default bucket currently?  I recall adding some additional logic for handling multiple concurrent applications with different scopes (e.g. Lockbox and Notes on the same Kinto instance) would that affect anything here?

Reply:
I will confirm with @vladikoff, but I think so, yes. The default bucket is our main tool for user isolation at the moment in Kinto.

Reply:
@vladikoff says it makes requests to https://testpilot.settings.services.mozilla.com/v1/buckets/default/collections/notes/, so yes, default buckets for everyone.